packages/components/src/components/hds/dropdown/index.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+
+import { HdsDropdownPositionValues } from './types.ts';
+
+import type { ComponentLike } from '@glint/template';
+import type { MenuPrimitiveSignature } from '../menu-primitive';
+import type { HdsDropdownFooterSignature } from './footer';
+import type { HdsDropdownHeaderSignature } from './header';
+import type { HdsDropdownListItemCheckboxSignature } from './list-item/checkbox';
+import type { HdsDropdownListItemCheckmarkSignature } from './list-item/checkmark';
+import type { HdsDropdownListItemCopyItemSignature } from './list-item/copy-item';
+import type { HdsDropdownListItemDescriptionSignature } from './list-item/description';
+import type { HdsDropdownListItemGenericSignature } from './list-item/generic';
+import type { HdsDropdownListItemInteractiveSignature } from './list-item/interactive';
+import type { HdsDropdownListItemRadioSignature } from './list-item/radio';
+import type { HdsDropdownListItemSeparatorSignature } from './list-item/separator';
+import type { HdsDropdownListItemTitleSignature } from './list-item/title';
+import type { HdsDropdownToggleButtonSignature } from './toggle/button';
+import type { HdsDropdownToggleIconSignature } from './toggle/icon';
+import type { HdsDropdownPositions } from './types';
+
+export const DEFAULT_POSITION = HdsDropdownPositionValues.BottomRight;
+export const POSITIONS: string[] = Object.values(HdsDropdownPositionValues);
+
+interface IndexSignature {
+  Args: MenuPrimitiveSignature['Args'] & {
+    height: string;
+    isInline: boolean;
+    listPosition: HdsDropdownPositions;
+    width: string;
+  };
+  Blocks: {
+    default: [
+      {
+        Footer?: ComponentLike<HdsDropdownFooterSignature>;
+        Header?: ComponentLike<HdsDropdownHeaderSignature>;
+        Checkbox?: ComponentLike<HdsDropdownListItemCheckboxSignature>;
+        Checkmark?: ComponentLike<HdsDropdownListItemCheckmarkSignature>;
+        CopyItem?: ComponentLike<HdsDropdownListItemCopyItemSignature>;
+        Description?: ComponentLike<HdsDropdownListItemDescriptionSignature>;
+        Generic?: ComponentLike<HdsDropdownListItemGenericSignature>;
+        Interactive?: ComponentLike<HdsDropdownListItemInteractiveSignature>;
+        Radio?: ComponentLike<HdsDropdownListItemRadioSignature>;
+        Separator?: ComponentLike<HdsDropdownListItemSeparatorSignature>;
+        Title?: ComponentLike<HdsDropdownListItemTitleSignature>;
+        ToggleButton?: ComponentLike<HdsDropdownToggleButtonSignature>;
+        ToggleIcon?: ComponentLike<HdsDropdownToggleIconSignature>;
+        close?: () => void;
```

Comment:
This `close` actually is coming from `MenuPrimitive` should we use its type instead? (I realize is much more verbose/complex, but is also technically more correct, no?)
```suggestion
        close?: MenuPrimitiveSignature['Blocks']['content']['close'];
```